src/Campaigns/Migrations/Donations/AddCampaignId.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace Give\Campaigns\Migrations\Donations;
+
+use Give\Donations\ValueObjects\DonationMetaKeys;
+use Give\Framework\Database\DB;
+use Give\Framework\Database\Exceptions\DatabaseQueryException;
+use Give\Framework\Migrations\Contracts\Migration;
+use Give\Framework\Migrations\Exceptions\DatabaseMigrationException;
+
+/**
+ * @unreleased
+ */
+class AddCampaignId extends Migration
+{
+    /**
+     * @inheritDoc
+     */
+    public static function id(): string
+    {
+        return 'add_campaign_id_to_donations';
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public static function title(): string
+    {
+        return 'Add campaign id to donations';
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public static function timestamp(): string
+    {
+        return strtotime('2024-11-22 00:00:00');
+    }
+
+    /**
+     * @inheritDoc
+     * @throws DatabaseMigrationException
+     */
+    public function run()
```

Comment:
I can't remember. Is this automatically run inside of a transaction? Or should we apply one here?

Reply:
Yep, see https://github.com/impress-org/givewp/blob/d056e04cbb1b43664b2b13c1df6fb7328cc0db11/src/Framework/Migrations/MigrationsRunner.php#L107